### ISSUE TITLE
Force is not honored if substitution rule is applied in Gradle 5.5.x

### ISSUE THREAD

**Issue by rpalcolea** [nebula-plugins/gradle-nebula-integration#63]
This is an issue involving forcing and substituting dependencies. 

Here are the result when running `./gradlew dependencyInsight --dependency test.nebula`

With Gradle 5.4.1, we see the following, which resolves to `1.0.0`:
```
> Task :dependencyInsight
test.nebula:a:1.0.0
   variant "compile" [
      org.gradle.status              = release (not requested)
      org.gradle.usage               = java-api
      org.gradle.category            = library (not requested)

      Requested attributes not found in the selected variant:
         org.gradle.dependency.bundling = external
         org.gradle.jvm.version         = 8
   ]
   Selection reasons:
      - Forced
      - Selected by rule : bad version

test.nebula:a:1.0.0
\--- compileClasspath

test.nebula:a:1.1.0 -> 1.0.0
\--- test.nebula:brings-a-transitively:0.15.0
     \--- compileClasspath (requested test.nebula:brings-a-transitively:latest.release)

test.nebula:brings-a-transitively:0.15.0
   variant "compile" [
      org.gradle.status              = release (not requested)
      org.gradle.usage               = java-api
      org.gradle.category            = library (not requested)

      Requested attributes not found in the selected variant:
         org.gradle.dependency.bundling = external
         org.gradle.jvm.version         = 8
   ]

test.nebula:brings-a-transitively:latest.release -> 0.15.0
\--- compileClasspath

```

With Gradle 5.5.1, we see the following, which resolves to `1.1.2`:
```
> Task :dependencyInsight
test.nebula:a:1.1.2
   variant "compile" [
      org.gradle.status              = release (not requested)
      org.gradle.usage               = java-api
      org.gradle.category            = library (not requested)

      Requested attributes not found in the selected variant:
         org.gradle.dependency.bundling = external
         org.gradle.jvm.version         = 8
   ]
   Selection reasons:
      - Forced
      - Selected by rule : bad version

test.nebula:a:1.0.0 -> 1.1.2
\--- compileClasspath

test.nebula:a:1.1.0 -> 1.1.2
\--- test.nebula:brings-a-transitively:0.15.0
     \--- compileClasspath (requested test.nebula:brings-a-transitively:latest.release)

test.nebula:brings-a-transitively:0.15.0
   variant "compile" [
      org.gradle.status              = release (not requested)
      org.gradle.usage               = java-api
      org.gradle.category            = library (not requested)

      Requested attributes not found in the selected variant:
         org.gradle.dependency.bundling = external
         org.gradle.jvm.version         = 8
   ]

test.nebula:brings-a-transitively:latest.release -> 0.15.0
\--- compileClasspath
```

Since `test.nebula:a` is forced to `1.0.0`, we expect gradle to use that instead of the result of the `resolutionStrategy.dependencySubstitution`, as it happened in previous versions.

Code can be found in https://github.com/nebula-plugins/gradle-nebula-integration/tree/master/force-with-substitution-break-sample

**Comment by ljacomet** [comment]
So this is a consequence of selector sorting that was introduced in Gradle 5.5 as a way to speed up resolution for locked configurations, among other things.

Looks like we should not do that sorting when there is a `forced` selector at play, otherwise we are changing the implicit rule that the first seen `force` wins to become the highest force wins.

Note that we plan to deprecate and eventually error when there are competing force and / or competing force and other substitutions as we believe these are problematic.

In the meantime, I'll work to have a fix for this in 5.6

**Comment by rpalcolea** [comment]
I confirmed that this is fixed in 5.6-rc-1, closing now

```
❯ ./gradlew dependencyInsight --dependency test.nebula
Starting a Gradle Daemon (subsequent builds will be faster)

> Task :dependencyInsight
test.nebula:a:1.0.0
   variant "compile" [
      org.gradle.status              = release (not requested)
      org.gradle.usage               = java-api
      org.gradle.libraryelements     = jar (compatible with: classes)
      org.gradle.category            = library (not requested)

      Requested attributes not found in the selected variant:
         org.gradle.dependency.bundling = external
         org.gradle.jvm.version         = 8
   ]
   Selection reasons:
      - Forced
      - Selected by rule : bad version

test.nebula:a:1.0.0
\--- compileClasspath

test.nebula:a:1.1.0 -> 1.0.0
\--- test.nebula:brings-a-transitively:0.15.0
     \--- compileClasspath (requested test.nebula:brings-a-transitively:latest.release)
```
